packages/bpk-component-info-banner/src/BpkInfoBannerInner.tsx

```diff
@@ -0,0 +1,228 @@
+    [ALERT_TYPES.SUCCESS]: getClassName('bpk-info-banner__success-icon'),
+    [ALERT_TYPES.WARNING]: getClassName('bpk-info-banner__warning-icon'),
+    [ALERT_TYPES.ERROR]: getClassName('bpk-info-banner__error-icon'),
+    [ALERT_TYPES.INFO]: getClassName('bpk-info-banner__info-icon'),
+  } as const;
+  const className = classMap[type];
+  const componentMap: {
+    [K in AlertTypeValue]: FunctionComponent<SVGProps<SVGSVGElement>>;
+  } = {
+    [ALERT_TYPES.SUCCESS]: TickCircleIcon,
+    [ALERT_TYPES.WARNING]: InfoCircleIcon,
+    [ALERT_TYPES.ERROR]: InfoCircleIcon,
+    [ALERT_TYPES.INFO]: InfoCircleIcon,
+  } as const;
+  const Icon = CustomIcon || componentMap[type];
+  const AlignedIcon = withButtonAlignment(Icon);
+
+  return <AlignedIcon className={className} />;
+};
+
+type ToggleButtonProps = {
+  label?: string;
+  expanded: boolean;
+};
+
+const ToggleButton = (props: ToggleButtonProps) => {
+  const classNames = [getClassName('bpk-info-banner__expand-icon')];
+  if (props.expanded) {
+    classNames.push(getClassName('bpk-info-banner__expand-icon--flipped'));
+  }
+
+  return (
+    <button
+      type="button"
+      className={getClassName('bpk-info-banner__toggle-button')}
+      aria-label={props.label}
+      aria-expanded={props.expanded}
+      title={props.label}
+    >
+      <ExpandIcon className={classNames.join(' ')} />
+    </button>
+  );
+};
+
+type Props = CommonProps & {
+  configuration?: (typeof CONFIGURATION)[keyof typeof CONFIGURATION];
+
+  // Only relevant when configuration == CONFIGURATION.EXPANDABLE
+  children?: ReactNode | string;
+  expanded?: boolean;
+  toggleButtonLabel?: string;
+  onExpandToggle?: OnExpandToggleHandler;
+
+  // Only relevant when configuration == CONFIGURATION.DISMISSABLE
+  dismissButtonLabel?: string;
+  onDismiss?: OnDismissHandler;
+};
+
+const BpkInfoBannerInner = ({
+  animateOnEnter = false,
+  animateOnLeave = false,
+  bannerClassName,
+  children = null,
+  configuration,
+  dismissButtonLabel = '',
+  expanded = false,
+  icon = null,
+  message,
+  onDismiss = null,
+  onExpandToggle = null,
+  show = true,
+  style = STYLE_TYPES.DEFAULT,
+  toggleButtonLabel = '',
+  type = ALERT_TYPES.INFO,
+  ...rest
+}: Props) => {
+  const onBannerExpandToggle = () => {
+    if (onExpandToggle) {
+      onExpandToggle(!expanded);
+    }
+  };
+
+  const onBannerDismiss = () => {
+    if (onDismiss) {
+      onDismiss();
+    }
+  };
+
+  const isExpandable = configuration === CONFIGURATION.EXPANDABLE;
+  const dismissable = configuration === CONFIGURATION.DISMISSABLE;
+  const showChildren = isExpandable && expanded;
+
+  const headerClassNames = [getClassName('bpk-info-banner__header')];
+  const sectionClassNames = [
+    'bpk-info-banner',
+    `bpk-info-banner--${type}`,
+    `bpk-info-banner--style-${style}`,
+  ].map((sectionClassName) => getClassName(sectionClassName));
+
+  if (bannerClassName) {
+    sectionClassNames.push(bannerClassName);
+  }
```

Comment:
```suggestion
  const sectionClassNames = [
    'bpk-info-banner',
    `bpk-info-banner--${type}`,
    `bpk-info-banner--style-${style}`,
  ].map((sectionClassName) => getClassName(
    sectionClassName
    bannerClassName && bannerClassName
  ));
```

Reply:
Won't it add a `bannerClassName` to each of the array elements? Original code adds it only once and these 2 pieces do not seem equivalent.
So, if `bannerClassName` provided via props, we'll have it in `sectionClassNames` 3 times.

Reply:
It shouldn't do unless my interpretation of this is narly 

The other suggestion we could do to make this simplified now I am thinking about it is this instead of mapping

```suggestion
  const sectionClassNames = getClassName(    
     'bpk-info-banner',
    `bpk-info-banner--${type}`,
    `bpk-info-banner--style-${style}`,
    bannerClassName && bannerClassName
    );
```

As type and style are always provided and we can just do this without needing logic

Reply:
Thanks!
Do we need to pass `bannerClassName` into `getClassName` function? Not really sure, but in original implementation it was added in the end. So, will it be much different if we'll concatenate `bannerClassName` to the result of this function instead of passing it as a param?